beamer/events.py

```diff
@@ -108,7 +108,15 @@ def _camel_to_snake(s: str) -> str:
 )
 
 
-def _make_topics_to_abi(contract: web3.contract.Contract) -> dict[bytes, ABIEvent]:
+def _make_topics_abi_mapping_for_contracts(contracts: list[Contract]) -> dict[bytes, ABIEvent]:
```

Comment:
I'd also use tuples here since we don't need mutability.
If one is in "proper Python typing" mode, I'd say using `Iterable[Contract]` would be most appropriate.